crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -827,14 +827,19 @@ impl<T: TransactionOrdering> TxPool<T> {
         tx_hash: &B256,
     ) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
         let (tx, pool) = self.all_transactions.remove_transaction_by_hash(tx_hash)?;
+
+        // After a tx is removed, its descendants must become parked due to the nonce gap
+        let updates = self.all_transactions.park_descendant_pending_txs(tx.id());
```

Comment:
I'd like this function to be a separate function because is remove_transaction_by_hash is called in two places:

https://github.com/paradigmxyz/reth/blob/main/crates/transaction-pool/src/pool/txpool.rs#L779-L786

and processing updates for this one is redundant.

Reply:
👍